### PR TITLE
feat: uplevel enabling of datastore and update of auth configs to top

### PR DESCRIPTION
**Pull request by kaustavghosh06**
*Description of changes:*
New amplify update api flow:

f45c89966b0d:datstore kaustavg$ amplify-dev update api
? Please select from one of the below mentioned services: GraphQL
? Select from the options below (Use arrow keys)
❯ Walkthrough all configurations 
  Update auth settings 
  Enable DataStore for entire API 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kaustavghosh06** [comment]
@yuth Yes, we should have a check for it but I would decouple this from control plane settings.  It should a part of the sanity check we do as a part of the gql-compile step and not when setting control plane configs - since the schema's are changed (based on which we should throw this error) and the only execution during a push is gql-compile.

File: packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js [hunk 1]
```diff
@@ -401,35 +401,73 @@ async function updateWalkthrough(context) {
...
+  let { updateOption } = await inquirer.prompt([updateOptionQuestion]);
 
-  if (amplifyMeta[category][resourceName].output.securityType) {
-    delete amplifyMeta[category][resourceName].output.securityType;
+  if (updateOption === 'enableDatastore') {
+    resolverConfig = {
+      project: { ConflictHandler: 'AUTOMERGE', ConflictDetection: 'VERSION' },
+    };
+  } else if (updateOption === 'authUpdate') {
+    /* eslint-disable */
```
Comment (yuth) [inline packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js]: Lets avoid disabling eslint rules inside documents

File: packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js [hunk 2]
```diff
@@ -401,35 +401,73 @@ async function updateWalkthrough(context) {
...
+    ({ authConfig, defaultAuthType } = await askDefaultAuthQuestion(context, parameters));
+    authConfig = await askAdditionalAuthQuestions(context, parameters, authConfig, defaultAuthType);
+    await checkForCognitoUserPools(context, parameters, authConfig);
+    /* eslint-disable */
```
Comment (yuth) [inline packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js]: same

File: packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js [hunk 3]
```diff
@@ -401,35 +401,73 @@ async function updateWalkthrough(context) {
...
-  const backendConfig = context.amplify.readJsonFile(backendConfigFilePath);
+    if (amplifyMeta[category][resourceName].output.securityType) {
+      delete amplifyMeta[category][resourceName].output.securityType;
+    }
 
-  if (backendConfig[category][resourceName].output.securityType) {
-    delete backendConfig[category][resourceName].output.securityType;
-  }
+    amplifyMeta[category][resourceName].output.authConfig = authConfig;
+    let jsonString = JSON.stringify(amplifyMeta, null, '\t');
```
Comment (yuth) [inline packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js]: Are we using tabs in all the places? We should stick to one, and I see in most of places, we use 4 spaces

Comment (kaustavghosh06) [reply packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js]: Done.

File: packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js (hunk 2 above)
Comment (kaustavghosh06) [reply packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js]: Removed.

File: packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js (hunk 1 above)
Comment (kaustavghosh06) [reply packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js]: Removed.

**Pull request merged**